packages/gatsby-image/src/index.js

```diff
@@ -384,33 +379,49 @@ class Image extends React.Component {
 
             {/* Once the image is visible, start downloading the image */}
             {this.state.isVisible && (
-              <Img
-                alt={alt}
-                title={title}
-                width={image.width}
-                height={image.height}
-                srcSet={image.srcSet}
-                src={image.src}
-                style={imageStyle}
-                onLoad={() => {
-                  this.setState({ imgLoaded: true })
-                  this.props.onLoad && this.props.onLoad()
-                }}
-              />
+              <picture>
+                {image.srcSetWebp && (<source
+                  type={`image/webp`}
+                  srcSet={image.srcSetWebp}
+                  sizes={image.sizes}
+                />)}
+
+                <source
+                  srcSet={image.srcSet}
+                  sizes={image.sizes}
+                />
+
+                <Img
+                  alt={alt}
+                  title={title}
+                  width={image.width}
+                  height={image.height}
+                  src={image.src}
+                  srcSet={image.srcSet}
+                  sizes={image.sizes}
+                  style={imageStyle}
+                  onLoad={() => {
```

Comment:
should `onLoad` event handler be used in `<source>` as well? or maybe it would work using it in `<picture>`?

Reply:
Img is a required tag in picture. It would seem to me that it would be redundant?